transaction-view/src/resolved_transaction_view.rs

```diff
@@ -0,0 +1,482 @@
+use {
+    crate::{transaction_data::TransactionData, transaction_view::TransactionView},
+    core::{
+        fmt::{Debug, Formatter},
+        ops::Deref,
+    },
+    solana_sdk::{
+        bpf_loader_upgradeable, ed25519_program,
+        hash::Hash,
+        message::{v0::LoadedAddresses, AccountKeys, TransactionSignatureDetails},
+        pubkey::Pubkey,
+        secp256k1_program,
+    },
+    solana_svm_transaction::{
+        instruction::SVMInstruction, message_address_table_lookup::SVMMessageAddressTableLookup,
+        svm_message::SVMMessage,
+    },
+    std::collections::HashSet,
+};
+
+/// A parsed and sanitized transaction view that has had all address lookups
+/// resolved.
+pub struct ResolvedTransactionView<D: TransactionData> {
+    /// The parsed and sanitized transction view.
+    view: TransactionView<true, D>,
+    /// The resolved address lookups.
+    resolved_addresses: LoadedAddresses,
+    /// A cache for whether an address is writable.
+    writable_cache: Vec<bool>, // TODO: should this be a vec, bitset, or array[256].
+}
+
+impl<D: TransactionData> Deref for ResolvedTransactionView<D> {
+    type Target = TransactionView<true, D>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.view
+    }
+}
+
+impl<D: TransactionData> ResolvedTransactionView<D> {
+    /// Given a parsed and sanitized transaction view, and a set of resolved
+    /// addresses, create a resolved transaction view.
+    pub fn new(
+        view: TransactionView<true, D>,
+        resolved_addresses: LoadedAddresses,
+        reserved_account_keys: &HashSet<Pubkey>, // why does this not use ahash at least?!
+    ) -> Self {
+        // verify that the number of readable and writable match up.
+        // This is a basic sanity check to make sure we're not passing a totally
+        // invalid set of resolved addresses.
+        assert_eq!(
+            resolved_addresses.writable.len(),
+            usize::from(view.total_writable_lookup_accounts())
+        );
+        assert_eq!(
+            resolved_addresses.readonly.len(),
+            usize::from(view.total_readonly_lookup_accounts())
+        );
```

Comment:
I think I prefer an assert here, but let me know what you think of my reasoning.

I generally treat Result/Option as an indication that "this may fail during normal operation - typically due to an issue with user input". But that is not the case here.
The only way this fails is if we screw up the arguments.

Admittedly that probably means the interface is not ideal.

Reply:
Yeah I think your reasoning makes perfect sense in the context of how we build the resolved transaction view in the runtime because immediately before this constructor we have a bunch of checks to make sure that address lookups all resolve correctly. But couldn't there be other contexts in which we or end users create this view?

Reply:
end users ideally aren't using this at all - it's meant to be an internal type, and we're really just exposing it because we need access within our own crates.

That said, I'd still argue that messing up the `LoadedAddreses` matching the `view` is an unrecoverable error that indicates a fundamental issue with the logic. In that case I still think an assert is preferable.

Reply:
but I think it's fine, we can just bubble that handling up - that's probably better?
It also may let us use simpler syntax:

```rust
let addresses = bank.load_addresses_ref(view.atls()).ok();
let resolved_transaction_view = ResolvedTransactionView::try_new(view, addresses, reserved_set);
```
instead of something similar to

```rust
let addresses = bank.load_addresses_ref(view.atls()).ok();
if let Some(addresses) = addresses {
    let resolved_transaction_view = ResolvedTransactionView::try_new(view, addresses, reserved_set);
}
```

Reply:
35f861cbb3bebbf6a5a12829395e28d3580a4631